pkg/obs/services/buckets.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package services
+
+import (
+	"fmt"
+	"github.com/huaweicloud/huaweicloud-csi-driver/pkg/config"
+	"github.com/huaweicloud/huaweicloud-sdk-go-obs/obs"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"net/http"
+)
+
+func GetBucket(c *config.CloudCredentials, bucketName string) (string, error) {
+	exists, err := CheckBucketExists(c, bucketName)
+	if err != nil {
+		return "", err
+	}
+	if !exists {
+		return "", nil
+	}
+	isParallelFile, err := CheckBucketFSStatus(c, bucketName)
+	if err != nil {
+		return "", err
+	}
+	if !isParallelFile {
+		return "", status.Errorf(codes.Unavailable, "Error, the OBS bucket %s is not a parallel file system", bucketName)
+	}
+	return bucketName, nil
+}
+
+func GetBucketMetadata(c *config.CloudCredentials, bucketName string) (*obs.GetBucketMetadataOutput, error) {
+	client, err := getObsClient(c)
+	if err != nil {
+		return nil, err
+	}
+	input := &obs.GetBucketMetadataInput{}
+	input.Bucket = bucketName
+	output, err := client.GetBucketMetadata(input)
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "Error, OBS instance %s does not exist, error: %v", bucketName, err)
```

Comment:
when err is not nil, if it is necessary to parse err message to verify it is 404?
for other err message, response code should be codes.Internal

Reply:
down